[android] Improve OSM profile UX

arnaudvergnet opened the pull request:
I felt the current way to log in to OSM was not very optimal so I simplified it. There is no page telling you to log in or register anymore. This page would take you to the log in screen if you click on a big "OpenStreetMap" button and felt unecessary. Instead, when making an edit or accessing the OSM profile while logged out, the user is directly taken to the login form, with the button to register moved here.

Other UX changes:
- Moved logout button out of the "more" bottom sheet in profile
- Added a progress indicator when loading user changes count
- Added a shortcut to the OSM profile in the main menu


Here is a video demonstrating the UX changes:

https://user-images.githubusercontent.com/80701113/177047342-63054b9b-7ce9-4bad-9570-4cab48d53d33.mp4

pastk commented:
Looks good at a first glance!

However I'm not sure whether its a good idea to add an OSM profile link into the pop-up menu. It makes sense to put there things that are used often. Why put OSM profile there?

arnaudvergnet commented:
> Looks good at a first glance!
> 
> However I'm not sure whether its a good idea to add an OSM profile link into the pop-up menu. It makes sense to put there things that are used often. Why put OSM profile there?

I though putting it here might help in making OSM known to new users. I also hope one day we will be able to have more information in the profile such as last edits, edits not yet published, manually publishing edits, ...

But this menu is very easy to add so if you want we can remove it and add it back later when the profile page has more information.

biodranik commented:
The main problem with your change is that it's not clear WHERE and WHY the user needs to log in.

biodranik commented:
Is it possible to login/register in OSM using Apple ID and/or Google login? Then we should consider these two options for the future too. Previously, the screen you removed had different login options.

We also plan to add a way to backup bookmarks. It will also require a login, probably a custom one, but maybe OSM auth can be reused, it will save us a lot of time.

arnaudvergnet commented:
> The main problem with your change is that it's not clear WHERE and WHY the user needs to log in.

I agree there is no mention of openstreetmap in the login form appart from the `register to osm` button. A fix would be to update the string `login_to_make_edits_visible` to `Login to OpenStreetMap so other users can see the changes that you have made from the string`.

biodranik commented:
> I agree there is no mention of openstreetmap in the login form appart from the `register to osm` button. A fix would be to update the string `login_to_make_edits_visible` to `Login to OpenStreetMap so other users can see the changes that you have made from the string`.

That what I meant by WHERE )
OpenStreetMap should be clickable and visually attractive. It should be very clear what is this login about.

pastk commented:
I've tested the change and it looks good to me!
(except the OSM profile link in the main menu pop-up)

Basically it removes an intermediate dialog screen which doesn't have any extra information and thus is redundant.

So, after a user taps "Save"...
Before: 1 => 2
After: 3
![osm-auth](https://user-images.githubusercontent.com/18434508/177536417-cc74c7db-8b23-4034-a50b-2a3c425e723d.png)

The situation is similar if going to the OSM profile page via settings: there were two redundant screens, now its one and no info is lost.

Minor things to improve:

Change the text to `Log into OpenStreetMap so other users can see the changes that you have made`.

Please add a "Don't have an OpenStreetMap.org account?" text above the register button (it was in the profile page before).

I think the text `We will check the changes. If we have any questions we will contact you via email.` is misleading.

![Screenshot_1657100458text](https://user-images.githubusercontent.com/18434508/177537089-fabf57d8-1d92-475b-b79e-ef8db09d90a7.png)

I suggest changing it to 
`OpenStreetMap editors may check the changes and get in contact if there are any questions.`

pastk commented:
And maybe change the `<- Log In` top bar to `<- Log into OpenStreetMap` and then maybe make OpenStreetMap in the text below bold / big / highlight color to make it super clear.

pastk commented:
I couldn't test how it looks after the login, 'cause the OSM auth seems to be broken again?
(doesn't work for me off the current master too).
@biodranik

biodranik commented:
@pastk double-checked, OSM login works for me on the master.

biodranik commented:
Let's make the login page clear for new users. Imagine a granny pressing the "Edit" button. She needs to register first (those who registered in OSM more or less know already what is it about). So we need a text (with links) like this:
> All map data comes from [OpenStreetMap.org](OpenStreetMap.org). It's like Wikipedia for maps where you can add or edit places, and they become available to millions of users all over the World.

arnaudvergnet commented:
> And maybe change the `<- Log In` top bar to `<- Log into OpenStreetMap` and then maybe make OpenStreetMap in the text below bold / big / highlight color to make it super clear.

I fear the text might be too long for some devices

> So we need a text (with links) like this:

I can edit the text in French and English, but what should I do with other languages?

arnaudvergnet commented:
I updated the strings for French and English and added a new one to explain what is OSM, to be displayed above the text `Log in so users...`.

I will also try to add the OSM logo at the top of the login page to make it even clearer.

biodranik commented:
We'll add other translations, it would be great to see how it looks in landscape and portrait.

arnaudvergnet commented:
> We'll add other translations, it would be great to see how it looks in landscape and portrait.

I have not pushed yet but here is the result. If you like it I will commit/push everything.

| Portrait | Landscape |
|-|-|
| ![Screenshot_1657376014](https://user-images.githubusercontent.com/80701113/178109588-b7581ded-e856-47da-b167-3da7f01f5861.png) | ![Screenshot_1657376019](https://user-images.githubusercontent.com/80701113/178109609-0fb67ced-b7a9-4c3a-b881-afd74022b6da.png) ![Screenshot_1657376026](https://user-images.githubusercontent.com/80701113/178109612-4ffb136a-eff2-4672-8ada-35de2c11af85.png) |

arnaudvergnet commented:
And here is for tablets:

![Screenshot_1657378092](https://user-images.githubusercontent.com/80701113/178110783-a7e6d2cb-4b72-4dbd-8678-fad74aa79c6c.png)

biodranik commented:
It would be great to have clickable links to OSM and to change text "Register at OpenStreetMap", so it will be clear for everyone.

arnaudvergnet commented:
Here, I think there is enough mention of osm now ;)

![Screenshot_1657400577](https://user-images.githubusercontent.com/80701113/178122557-bbd80eb8-7ac5-4304-b6e0-b31f2e9eb251.png)

biodranik commented:
Aargh, now scrolling is needed…

arnaudvergnet commented:
This screenshot was taken on a Pixel 2 which has a small screen. The scroll here is not a big deal because this screen is not used often. At least here you are sure users will see the text about osm.

biodranik commented:
There are many older smaller devices. In this case, it should be very clear for users that there is a scroll because the REGISTER button won't be visible otherwise. It's better to leave it visible.

arnaudvergnet commented:
Showing scrollbars is a good idea, here is how it looks on small screens:

| Portrait | Landscape |
|-|-|
| ![Screenshot_1657449142](https://user-images.githubusercontent.com/80701113/178141258-ee1aa52b-dc85-4b29-9784-d143ae607684.png) | ![Screenshot_1657449155](https://user-images.githubusercontent.com/80701113/178141260-680a4705-9fbf-4073-8c08-2a05f58107ec.png) |

biodranik commented:
What if
1. Make OSM icon smaller 1.5-2x times.
2. Put icon on the left for the portrait mode
3. Use one line for login and password for portrait mode?

The idea is to fit it somehow into ANY screen and avoid scrolling (we want as simple UX as possible, right? With the minimum number of clicks). Ideally with autoscaling (e.g. font and icons may be smaller, but everything will be still visible).

arnaudvergnet commented:
No scroll on the Pixel 2 anymore. Scroll might still happen on some smaller phones but scrollviews exist for a reason ;)

| Portrait | Landscape |
|-|-|
| ![Screenshot_1657458132](https://user-images.githubusercontent.com/80701113/178146003-4bb2fc8e-795e-4fc2-aa3a-5f5bb38a54d5.png) | ![Screenshot_1657458136](https://user-images.githubusercontent.com/80701113/178146005-b618bd06-ee25-40e2-aa1d-15a52f1f7b50.png) |

biodranik commented:
Cool! Two more changes and the landscape would be perfect ;-)
1. One line for Login and forgot password (and dynamically reduce the font size if the landscape is not too wide).
2. One line for "Don't have..?" and Register at OpenStreetMap

arnaudvergnet commented:
Is it better?

![Screenshot_1657694915](https://user-images.githubusercontent.com/80701113/178669330-1070a01d-cb4d-4374-80d2-ac0c402c663e.png)

biodranik commented:
Oh yeah, that's better! Does portrait still look good?

rtsisyk reviewed (approved):
The new UI is better than previous version. Looks good to me.

arnaudvergnet commented:
@biodranik is everything good for you or do you have other changes in mind?

biodranik commented on android/src/com/mapswithme/maps/editor/OsmLoginFragment.java:
```diff
@@ -0,0 +1,127 @@
+  {
+    OsmOAuth.setAuthorization(requireContext(), auth[0], auth[1], username);
+    Bundle extras = requireActivity().getIntent().getExtras();
+    int extraOnSuccess = extras.getInt("onSuccess");
```
nit: final here and above.

biodranik commented on android/src/com/mapswithme/maps/editor/OsmLoginFragment.java:
```diff
@@ -0,0 +1,127 @@
+
+  public enum OnSuccess
+  {
+    profile
```
Is this enum with one value really needed? :)

biodranik commented on data/strings/strings.txt:
```diff
@@ -10489,6 +10489,11 @@
+  [login_osm]
+    tags = android
+    en = Log in to OpenStreetMap
+    fr = Se connecter à OpenStreetMap
```
en = Login to OpenStreetMap
ar = OpenStreetMap ﻰﻟﺇ ﻝﻮﺧﺪﻟﺍ ﻞﻴﺠﺴﺗ
be = Увайсці ў OpenStreetMap
bg = Влезте в OpenStreetMap
cs = Přihlaste se do OpenStreetMap
da = Log ind på OpenStreetMap
de = Melden Sie sich bei OpenStreetMap an
el = Συνδεθείτε στο OpenStreetMap
es = Iniciar sesión en OpenStreetMap
eu = Hasi saioa OpenStreetMap-en
fa = ﺪﯾﻮﺷ OpenStreetMap ﺩﺭﺍﻭ
fi = Kirjaudu OpenStreetMapiin
fr = Connectez-vous à OpenStreetMap
he = OpenStreetMap-ל סנכיה
hu = Jelentkezzen be az OpenStreetMap-be
id = Masuk ke OpenStreetMap
it = Accedi a OpenStreetMap
ja = OpenStreetMapにログイン
ko = OpenStreetMap에 로그인
nb = Logg på OpenStreetMap
nl = Inloggen op OpenStreetMap
pl = Zaloguj się do OpenStreetMap
pt = Entrar no OpenStreetMap
pt-BR = Entrar no OpenStreetMap
ro = Conectați-vă la OpenStreetMap
ru = Войти в OpenStreetMap
sk = Prihláste sa do OpenStreetMap
sv = Logga in på OpenStreetMap
sw = Ingia kwa OpenStreetMap
th = เข้าสู่ระบบ OpenStreetMap
tr = OpenStreetMap'e giriş yapın
uk = Увійдіть до OpenStreetMap
vi = Đăng nhập vào OpenStreetMap
zh-Hans = 登录 OpenStreetMap
zh-Hant = 登錄 OpenStreetMap

biodranik commented on data/strings/strings.txt:
```diff
@@ -12231,9 +12236,14 @@
 
+  [osm_presentation]
+    tags = android
+    en = All map data comes from the OpenStreetMap project. It is like Wikipedia for maps where you can add or edit places, and they become available to millions of users all over the World.
```
en = All map data comes from the OpenStreetMap project. It is like Wikipedia for maps where you can add or edit places, and they become available to millions of users all over the World.
ar = .ﻢﻟﺎﻌﻟﺍ ءﺎﺤﻧﺃ ﻊﻴﻤﺟ ﻲﻓ ﻦﻴﻣﺪﺨﺘﺴﻤﻟﺍ ﻦﻴﻳﻼﻤﻟ ﺔﺣﺎﺘﻣ ﺢﺒﺼﺗﻭ ، ﺎﻬﻠﻳﺪﻌﺗ ﻭﺃ ﻦﻛﺎﻣﻷﺍ ﺔﻓﺎﺿﺇ
be = Усе дадзеныя карты паходзяць з праекта OpenStreetMap. Гэта як Wikipedia для карт, дзе вы можаце дадаваць або рэдагаваць месцы, і яны становяцца даступнымі для мільёнаў карыстальнікаў па ўсім свеце.
bg = Всички картографски данни идват от проекта OpenStreetMap. Това е като Wikipedia за карти, където можете да добавяте или редактирате места и те стават достъпни за милиони потребители по целия свят.
cs = Všechna mapová data pocházejí z projektu OpenStreetMap. Je to jako Wikipedie s mapami, kde můžete přidávat nebo upravovat místa a jsou dostupné milionům uživatelů po celém světě.
da = Alle kortdata kommer fra OpenStreetMap-projektet. Det er ligesom Wikipedia for kort, hvor du kan tilføje eller redigere steder, og de bliver tilgængelige for millioner af brugere over hele verden.
de = Alle Kartendaten stammen aus dem OpenStreetMap-Projekt. Es ist wie Wikipedia für Karten, wo Sie Orte hinzufügen oder bearbeiten können, und sie werden Millionen von Benutzern auf der ganzen Welt zur Verfügung gestellt.
el = Όλα τα δεδομένα χάρτη προέρχονται από το έργο OpenStreetMap. Μοιάζει με τη Wikipedia για χάρτες όπου μπορείτε να προσθέσετε ή να επεξεργαστείτε μέρη και γίνονται διαθέσιμα σε εκατομμύρια χρήστες σε όλο τον κόσμο.
es = Todos los datos del mapa provienen del proyecto OpenStreetMap. Es como Wikipedia para mapas donde puedes agregar o editar lugares, y están disponibles para millones de usuarios en todo el mundo.
eu = Maparen datu guztiak OpenStreetMap proiektutik datoz. Tokiak gehitu edo edita ditzakezun mapetarako Wikipedia bezalakoa da, eta mundu osoko milioika erabiltzaileren eskura jartzen dira.
fa = .ﺪﻧﺮﯿﮔu200cﯽﻣ ﺭﺍﺮﻗ ﺱﺮﺘﺳﺩ ﺭﺩ ﻥﺎﻬﺟ ﺮﺳﺍﺮﺳ ﺭﺩ ﺮﺑﺭﺎﮐ ﺎﻫu200cﻥﻮﯿﻠﯿﻣ ﯼﺍﺮﺑ ﺎﻫu200cﻥﺁ ﻭ
fi = Kaikki karttatiedot tulevat OpenStreetMap-projektista. Se on kuin karttojen Wikipedia, jossa voit lisätä tai muokata paikkoja, ja ne ovat miljoonien käyttäjien saatavilla kaikkialla maailmassa.
fr = Toutes les données cartographiques proviennent du projet OpenStreetMap. C'est comme Wikipedia pour les cartes où vous pouvez ajouter ou modifier des lieux, et ils deviennent disponibles pour des millions d'utilisateurs partout dans le monde.
he = .םלועה יבחר לכב םישמתשמ ינוילימל תונימזל תוכפוה ןהו ,תומוקמ ךורעל וא ףיסוהל לו
hu = Minden térképadat az OpenStreetMap projektből származik. Olyan ez, mint a Wikipédia a térképekhez, ahol helyeket adhat hozzá vagy szerkeszthet, és a felhasználók milliói számára válnak elérhetővé szerte a világon.
id = Semua data peta berasal dari proyek OpenStreetMap. Ini seperti Wikipedia untuk peta tempat Anda dapat menambahkan atau mengedit tempat, dan peta tersebut tersedia bagi jutaan pengguna di seluruh Dunia.
it = Tutti i dati della mappa provengono dal progetto OpenStreetMap. È come Wikipedia per le mappe in cui puoi aggiungere o modificare luoghi e diventano disponibili per milioni di utenti in tutto il mondo.
ja = すべての地図データは、OpenStreetMap プロジェクトから取得されます。場所を追加または編集できる地図のウィキペディアのようなもので、世界中の何百万人ものユーザーが利用できるようになります。
ko = 모든 지도 데이터는 OpenStreetMap 프로젝트에서 가져옵니다. ""장소를 추가하거나 편집할 수 있는 지도용 Wikipedia와 같으며 전 세계 수백만 사용자가 사용할 수 있습니다.
nb = Alle kartdata kommer fra OpenStreetMap-prosjektet. Det er som Wikipedia for kart hvor du kan legge til eller redigere steder, og de blir tilgjengelige for millioner av brukere over hele verden.
nl = Alle kaartgegevens komen uit het OpenStreetMap-project. Het is als Wikipedia voor kaarten waar je plaatsen kunt toevoegen of bewerken, en ze worden beschikbaar voor miljoenen gebruikers over de hele wereld.
pl = Wszystkie dane mapy pochodzą z projektu OpenStreetMap. Przypomina Wikipedię z mapami, w których można dodawać lub edytować miejsca, a stają się one dostępne dla milionów użytkowników na całym świecie.
pt = Todos os dados cartográficos provêm do projecto OpenStreetMap. É como a Wikipedia para mapas onde se podem adicionar ou editar locais, e estes ficam disponíveis para milhões de utilizadores em todo o mundo.
pt-BR = Todos os dados do mapa são provenientes do projeto OpenStreetMap. É como a Wikipedia para mapas onde você pode adicionar ou editar lugares, e eles ficam disponíveis para milhões de usuários em todo o mundo.
ro = Toate datele hărții provin din proiectul OpenStreetMap. Este ca Wikipedia pentru hărți în care puteți adăuga sau edita locuri și acestea devin disponibile pentru milioane de utilizatori din întreaga lume.
ru = Все данные карты поступают из проекта OpenStreetMap. Это как Википедия для карт, где вы можете добавлять или редактировать места, и они становятся доступными для миллионов пользователей по всему миру.
sk = Všetky mapové podklady pochádzajú z projektu OpenStreetMap. Je to ako Wikipedia pre mapy, kde môžete pridávať alebo upravovať miesta a sú dostupné pre milióny používateľov na celom svete.
sv = All kartdata kommer från OpenStreetMap-projektet. Det är som Wikipedia för kartor där du kan lägga till eller redigera platser, och de blir tillgängliga för miljontals användare över hela världen.
sw = Data yote ya ramani inatoka kwa mradi wa OpenStreetMap. Ni kama Wikipedia ya ramani ambapo unaweza kuongeza au kuhariri maeneo, na yanapatikana kwa mamilioni ya watumiaji duniani kote.
th = ข้อมูลแผนที่ทั้งหมดมาจากโครงการ OpenStreetMap เหมือนกับวิกิพีเดียสำหรับแผนที่ที่คุณสามารถเพิ่มหรือแก้ไขสถานที่ได้ และพวกเขาพร้อมให้ผู้ใช้หลายล้านคนทั่วโลกใช้งานได้
tr = Tüm harita verileri OpenStreetMap projesinden gelir. Yer ekleyebileceğiniz veya düzenleyebileceğiniz haritalar için Wikipedia gibidir ve bunlar tüm dünyada milyonlarca kullanıcı tarafından kullanılabilir hale gelir.
uk = Усі картографічні дані надходять із проекту OpenStreetMap. Це як Wikipedia для карт, де ви можете додавати або редагувати місця, і вони стають доступними для мільйонів користувачів у всьому світі.
vi = Tất cả dữ liệu bản đồ đến từ dự án OpenStreetMap. Nó giống như Wikipedia dành cho bản đồ nơi bạn có thể thêm hoặc chỉnh sửa các địa điểm và chúng có sẵn cho hàng triệu người dùng trên khắp Thế giới.
zh-Hans = 所有地图数据均来自 OpenStreetMap 项目。它就像 Wikipedia 中的地图，您可以在其中添加或编辑地点，并且它们可供全球数百万用户使用。
zh-Hant = 所有地圖數據均來自 OpenStreetMap 項目。它就像 Wikipedia 中的地圖，您可以在其中添加或編輯地點，並且它們可供全球數百萬用戶使用。

biodranik commented on data/strings/strings.txt:
```diff
@@ -12231,9 +12236,14 @@
   [login_to_make_edits_visible]
     tags = android,ios
-    en = Log in so other users can see the changes that you have made
+    en = Log in to openstreetmap.org so other users can see the changes you have made.
```
```suggestion
    en = Log in to OpenStreetMap.org so other users can see the changes you have made.
```

biodranik commented on data/strings/strings.txt:
```diff
@@ -12231,9 +12236,14 @@
-    en = Log in so other users can see the changes that you have made
+    en = Log in to openstreetmap.org so other users can see the changes you have made.
     ar = قم بتسجيل الدخول حتى يتمكن باقي المستخدمين من رؤية التغييرات التي قمت بها.
     be = Аўтарызуйцеся, каб іншыя карыстальнікі бачылі вашыя змены.
```
```suggestion
    be = Аўтарызуйцеся ў OpenStreetMap.org, каб іншыя карыстальнікі бачылі вашыя змены.
```

biodranik replied:
ru = Авторизуйтесь в OpenStreetMap.org, чтобы ваши изменения увидели другие пользователи.

biodranik replied:
uk = Увійдіть у OpenStreetMap.org, щоб ваші зміни побачили інші користувачі.

biodranik commented on data/strings/strings.txt:
```diff
@@ -14616,7 +14626,7 @@
   [editor_share_to_all_dialog_message_2]
     tags = android,ios
-    en = We will check the changes. If we have any questions we will contact you via email.
+    en = OpenStreetMap editors will check the changes and get in contact with you if they have any questions.
```
en = OpenStreetMap editors will check the changes and get in contact with you if they have any questions.
ar =  .ﺔﻠﺌﺳﺃ ﻱﺃ ﻢﻬﻳﺪﻟ ﻥﺎﻛ ﺍﺫﺇ ﻚﻌﻣ ﻥﻮﻠﺻﺍﻮﺘﻳﻭ ﺕﺍﺮﻴﻴﻐﺘﻟﺍ ﻦﻣ OpenStreetMap ﻭﺭﺮﺤﻣ ﻖﻘﺤﺘﻴﺳ
be = Рэдактары OpenStreetMap правераць змены і звяжуцца з вамі, калі ў іх узнікнуць пытанні.
bg = Редакторите на OpenStreetMap ще проверят промените и ще се свържат с вас, ако имат въпроси.
cs = Redaktoři OpenStreetMap zkontrolují změny a budou vás kontaktovat, pokud budou mít nějaké dotazy.
da = OpenStreetMap-redaktører vil tjekke ændringerne og kontakte dig, hvis de har spørgsmål.
de = OpenStreetMap-Redakteure werden die Änderungen prüfen und sich bei Fragen mit Ihnen in Verbindung setzen.
el = Οι συντάκτες του OpenStreetMap θα ελέγξουν τις αλλαγές και θα επικοινωνήσουν μαζί σας εάν έχουν απορίες.
es = Los editores de OpenStreetMap comprobarán los cambios y se pondrán en contacto contigo si tienen alguna pregunta.
eu = OpenStreetMap editoreek aldaketak egiaztatuko dituzte eta zurekin harremanetan jarriko dira zalantzaren bat izanez gero.
fa = .ﺖﻓﺮﮔ ﺪﻨﻫﺍﻮﺧ ﺱﺎﻤﺗ ﺎﻤﺷ ﺎﺑ ﻝﺍﻮﺳ ﻪﻧﻮﮔﺮﻫ ﻦﺘﺷﺍﺩ ﺕﺭﻮﺻ ﺭﺩ ﻭ ﻩﺩﺮﮐ ﯽﺳﺭﺮﺑ ﺍﺭ ﺕﺍﺮﯿﯿﻐﺗ Ope
fi = OpenStreetMap-editorit tarkistavat muutokset ja ottavat sinuun yhteyttä, jos heillä on kysyttävää.
fr = Les éditeurs d'OpenStreetMap vérifieront les modifications et vous contacteront s'ils ont des questions.
he = .ןהשלכ תולאש םהל שי םא רשק ךתיא ורציו םייונישה תא וקדבי OpenStreetMap יכרוע
hu = Az OpenStreetMap szerkesztői ellenőrzik a változásokat, és felveszik Önnel a kapcsolatot, ha kérdéseik vannak.
id = Editor OpenStreetMap akan memeriksa perubahan dan menghubungi Anda jika ada pertanyaan.
it = Gli editori di OpenStreetMap controlleranno le modifiche e si metteranno in contatto con te in caso di domande.
ja = OpenStreetMap の編集者が変更を確認し、質問があれば連絡します。
ko = OpenStreetMap 편집자가 변경 사항을 확인하고 질문이 있는 경우 귀하에게 연락할 것입니다.
nb = OpenStreetMap-redaktører vil sjekke endringene og ta kontakt med deg hvis de har spørsmål.
nl = OpenStreetMap-editors zullen de wijzigingen controleren en contact met u opnemen als ze vragen hebben.
pl = Redaktorzy OpenStreetMap sprawdzą zmiany i skontaktują się z Tobą, jeśli będą mieli jakiekolwiek pytania.
pt = Os editores do OpenStreetMap verificarão as alterações e entrarão em contacto consigo se tiverem alguma dúvida.
pt-BR = Os editores do OpenStreetMap verificarão as mudanças e entrarão em contato com você se tiverem alguma dúvida.
ro = Editorii OpenStreetMap vor verifica modificările și vor lua legătura cu dvs. dacă au întrebări.
ru = Редакторы OpenStreetMap проверят изменения и свяжутся с вами, если у них возникнут вопросы.
sk = Redaktori OpenStreetMap skontrolujú zmeny a budú vás kontaktovať, ak budú mať nejaké otázky.
sv = OpenStreetMap-redigerare kommer att kontrollera ändringarna och kontakta dig om de har några frågor.
sw = Wahariri wa OpenStreetMap watakagua mabadiliko na kuwasiliana nawe ikiwa wana maswali yoyote.
th = ผู้แก้ไข OpenStreetMap จะตรวจสอบการเปลี่ยนแปลงและติดต่อคุณหากมีคำถามใดๆ
tr = OpenStreetMap editörleri değişiklikleri kontrol edecek ve soruları olursa sizinle iletişime geçecektir.
uk = Редактори OpenStreetMap перевірять зміни та зв’яжуться з вами, якщо у них виникнуть запитання.
vi = Các biên tập viên của OpenStreetMap sẽ kiểm tra các thay đổi và liên hệ với bạn nếu họ có bất kỳ câu hỏi nào.
zh-Hans = OpenStreetMap 编辑人员将检查更改并在有任何问题时与您联系。
zh-Hant = OpenStreetMap 編輯人員將檢查更改並在有任何問題時與您聯繫。

biodranik reviewed (changes requested):
We're almost there!

arnaudvergnet commented:
I rebased the PR on master instead of merging so the history is easier to read. I will try to reorder/clean up the commits as well.

arnaudvergnet commented:
@biodranik rebased on master to fix conflicts. Catalan was added since I started this PR so it is missing in added strings.

rtsisyk commented:
Merged. Thanks for this good rework of OSM login page.

muralito commented:
The users shouldn't put their OSM credentials in Organic Maps. The login should be made via Oauth/Oauth2 directly in the OSM website, and there authorize Organic Maps to upload/modify data, just like JOSM or StreetComplete does it.

biodranik commented:
@muralito good point, but it may require a lot of work. Will you create an issue?
For some reason, it didn't go several years ago in mapsme. @Zverik do you remember why?

Zverik commented:
Google had an issue with webview authorization, I think. And by the way, mapsme did not store user credentials, but emulated an oauth flow and stored an oauth token. 

20 сентября 2022 г. 02:12:17 GMT+03:00, Alexander Borsuk ***@***.***> пишет:
***@***.*** good point, but it may require a lot of work. Will you create an issue?
>For some reason, it didn't go several years ago in mapsme. @Zverik do you remember why?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/organicmaps/organicmaps/pull/2909#issuecomment-1251662092
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

biodranik commented:
@Zverik Thanks! You're right, I checked the code: Organic Maps does not save password, only user name, and tokens.
